mailpoet/lib/Automation/Engine/Registry.php

```diff
@@ -33,11 +37,31 @@ public function getSubjects(): array {
     return $this->subjects;
   }
 
+  public function addStep(Step $step): void {
+    if ($step instanceof Trigger) {
+      $this->addTrigger($step);
+    } elseif ($step instanceof Action) {
+      $this->addAction($step);
+    }
+
+    // TODO: allow adding any other step implementations?
+  }
```

Comment:
I probably added this only for some kind of completeness and symmetry with getters.

The whole process that led me to adding the parent `Step` was that originally, instead of the `WORKFLOW_STEP_BEFORE_SAVE` hook, I was considering calling sth like `$step->onBeforeSave()` but then I realized we may end up with a huge interface over time and that some kind of events would be better and since we're in WP then actions/filters would be the most idiomatic. Having this done, I wanted to keep it because I think having the parent `Step` is correct in any way.